tests/standalone_display_test.cpp

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (C) 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+// #define LOG_NDEBUG 0 // Uncomment to see HWC2 API calls in logcat
+
+#define LOG_TAG "standalone_display_test"
+
+#include <xf86drm.h>
+
+#include <algorithm>
+#include <cinttypes>
+#include "DrmHwcTwo.h"
+#include "HwcLayer.h"
+#include "backend/Backend.h"
+#include "utils/log.h"
+#include <cutils/properties.h>
+#include <string.h>
+#include <sys/mman.h>
+
+#define MAX_PIPE_NUM 4
+#define MAX_PLANE_NUM 7
+#define MAX_SCALING_NUM 2
+#define MAX_LAYER_NUM MAX_PLANE_NUM
+
+namespace android {
+class DrmHwcTwoTest : public DrmHwcTwo {
+ public:
+  DrmHwcTwoTest() {
+    GetResMan().Init();
+  }
+  ~DrmHwcTwoTest() override = default;
+};
+struct Drmhwc2DeviceTest : hwc2_device {
+  DrmHwcTwoTest drmhwctwotest;
+};
+}  // namespace android
+
```

Comment:
it is better you add one fill_buffer function, then put all related code in that function.